### PR TITLE
Setup and apply prettier

### PR DESCRIPTION
**Pull request by murdos**

### CODE REVIEW

File: generators/app/templates/generators/client/_index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /* eslint-disable consistent-return */
-const chalk = require("chalk");
-const ClientGenerator = require("generator-jhipster/generators/client");
-const prompts = require("./prompts");
-const writeFiles = require("./files").writeFiles;
+const chalk = require('chalk');
```
Comment (deepu105) [inline generators/app/templates/generators/client/_index.js]: while the applied formatting is fine, we should not run prettier on the actual template files and they should be excluded in prettierignore. Instead we should setup prettier transform on the templates while generating them like in jhipster

**Comment by deepu105** [comment]
Thanks btw

File: generators/app/templates/generators/client/_index.js (hunk 1 above)
Comment (murdos) [reply generators/app/templates/generators/client/_index.js]: Alright, that sounds indeed a better idea in the long term :)
I'll check how to do that. I guess something like that is done in generator-jhipster.

Comment (deepu105) [reply generators/app/templates/generators/client/_index.js]: yes its already done that way in JHipster, look for the prettier setup of generator, generated project and `generator-transform.js`

**Comment by murdos** [comment]
PR updated: I've setup prettier transform on the templates while generating, and also added prettier dependencies and config to the generated module, so everything is aligned.

File: package.json [hunk 2]
```diff
@@ -23,13 +23,18 @@
   },
   "dependencies": {
     "chalk": "2.4.1",
+    "gulp-filter": "^5.1.0",
```
Comment (deepu105) [inline package.json]: Can you afd only exact version everywhere, we don't use relative versions

**Comment by murdos** [comment]
Requested change done ;-)

**Pull request merged**